modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/defragmentation/CachePartitionDefragmentationManager.java

```diff
@@ -1072,61 +1114,76 @@ public synchronized void onCacheGroupFinish(CacheGroupContext grpCtx) {
         }
 
         /** */
-        public synchronized void onCacheGroupSkipped(CacheGroupContext grpCtx) {
+        private synchronized void onCacheGroupSkipped(CacheGroupContext grpCtx, int partitions) {
             scheduledGroups.remove(grpCtx.cacheOrGroupName());
 
             skippedGroups.add(grpCtx.cacheOrGroupName());
+
+            defragmentedPartitionCount += partitions;
         }
 
         /** */
-        public synchronized void onFinish() {
+        private synchronized void onFinish() {
             finishTs = System.currentTimeMillis();
 
             log.info("Defragmentation process completed. Time: " + (finishTs - startTs) * 1e-3 + "s.");
         }
 
-        /** {@inheritDoc} */
-        @Override public synchronized String toString() {
-            StringBuilder sb = new StringBuilder();
-
-            if (!finishedGroups.isEmpty()) {
-                sb.append("Defragmentation is completed for cache groups:\n");
+        /** */
+        private synchronized Status copy() {
+            return new Status(
+                startTs,
+                finishTs,
+                new HashSet<>(scheduledGroups),
+                new HashMap<>(progressGroups),
+                new HashMap<>(finishedGroups),
+                new HashSet<>(skippedGroups)
+            );
+        }
 
-                for (Map.Entry<CacheGroupContext, DefragmentationCacheGroupProgress> entry : finishedGroups.entrySet()) {
-                    sb.append("    ").append(entry.getKey().cacheOrGroupName()).append(" - ");
+        /** */
+        public long getStartTs() {
```

Comment:
Auto-generated names? Can you please change them? We don't use abbreviations in method names.